Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+        private static readonly ISet<char> EscapeLiterals = new HashSet<char>();
+        private readonly string specifier;
+
+        static Literal() {
+            foreach (char escape in new char[]{ '.', '+', '*', '?', '(', ')', '{', '}', '[', ']', '|', '\\' })
+            {
+                EscapeLiterals.Add(escape);
+            }
+        }
+
+        public Literal(string specifier)
+        {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a Literal");
+            }
+            this.specifier = specifier;
+        }
+
+        public string Specifier
+        {
+            get
+            {
+                return specifier;
+            }
+        }
+
+        public string Description
+        {
+            get
+            {
+                // here be dragons!
+                // keep track of:
+                // - escaped chars
+                // - escape sequences (each having a different description)
+                // - codepoint escapes (belongs into above category but kept separate)
+                // - and actually boring literal matches
+                if (specifier.Length > 1)
+                {
+                    string relevant = specifier.Substring(1); // skip the damn Backslash at the start
+                    if (relevant.Length > 1) // longer sequences
+                    {
+                        if (relevant.StartsWith("u"))
+                        {
+                            return string.Format(AssistantResources.AtomDescription_Literal_UnicodePoint, relevant.Substring(1)); //skip u
+                        }
+                        else if (relevant.StartsWith("x"))
+                        {
+                            return string.Format(AssistantResources.AtomDescription_Literal_HexCodepoint, relevant.Substring(1)); // skip x
+                        }
+                        else
+                        {
+                            return string.Format(AssistantResources.AtomDescription_Literal_OctalCodepoint, relevant); // no format specifier to skip
+                        }
+                    }
+                    else if (EscapeLiterals.Contains(relevant[0]))
+                    {
+                        return string.Format(AssistantResources.AtomDescription_Literal_EscapedLiteral, relevant);
+                    }
+                    else if (char.IsDigit(relevant[0]))
+                    {
+                        return string.Format(AssistantResources.AtomDescription_Literal_Backreference, relevant);
+                    }
+                    else
+                    {
+                        // special escapes here
+                        switch (relevant[0])
```

Comment:
I think I'd make a `Dictionary` for this. Be it only to reduce the size of the method!